modules/pam_faillock/main.c

```diff
@@ -51,29 +51,37 @@
 #define AUDIT_NO_ID     ((unsigned int) -1)
 #endif
 
+#include "pam_inline.h"
 #include "faillock.h"
-
-struct options {
-	unsigned int reset;
-	const char *dir;
-	const char *user;
-	const char *progname;
-};
+#include "faillock_config.h"
 
 static int
 args_parse(int argc, char **argv, struct options *opts)
 {
 	int i;
+	int rv;
+	const char *conf = default_faillock_conf;
+
 	memset(opts, 0, sizeof(*opts));
 
-	opts->dir = FAILLOCK_DEFAULT_TALLYDIR;
+	opts->dir = strdup(FAILLOCK_DEFAULT_TALLYDIR);
 	opts->progname = argv[0];
 
 	for (i = 1; i < argc; ++i) {
-		if (strcmp(argv[i], "--dir") == 0) {
+		if (strcmp(argv[i], "--conf") == 0) {
 			++i;
 			if (i >= argc || strlen(argv[i]) == 0) {
-				fprintf(stderr, "%s: No directory supplied.\n", argv[0]);
+				fprintf(stderr, "%s: No configuration directory supplied.\n",
+						argv[0]);
+				return -1;
+			}
+			conf = pam_str_skip_prefix(argv[i], "--conf=");
```

Comment:
This does not look correct. Why would you skip the prefix? There should be none.

Reply:
You are right. I fixed it.